README.md

```diff
@@ -13,22 +13,167 @@
+type Value union {
+  | Boolean
+  | Integer
+  | Float
+  | String 
+  | Link
+  | Bytes
+} representation kinded
+```
+ 
+Note that all floating point values MUST be representable as 64-bit (double-precision) floats as defined in [IEEE 754-2019] when deserialized. `NaN`s MUST NOT be used.
+
+### 2.1.5 Caused By
+
+Links to other facts MUST be placed in an array in the `causedBy` field.
+
+``` ipldsch
+type CausedBy = [&Fact]
+```
+ 
+## 2.1.6 Capsule
+
+An OPTIONAL capsule type to clarify the enclosed data MAY be used.
+
+``` ipldsch
+type FactCapsule struct {
+  c Fact (rename "pomodb/v0.1/fact")
+}
+```
+
+## 2.2 Store
+
+As PomoDB is intended to operate over many Facts, storing and referencing collections is important. Below are two strategies for representing groups of Facts.
+
+### 2.2.1 Hash Map
+
+A collection of Facts is called a "Store". This structure is simply a content-addressed blockstore indexing Facts by their CID.
+
+``` ipldsch
+type Store = {CID : Fact}
+```
+
+### 2.2.2 Set
+
+Stores contain too much information for transmission and encrypted storage. In these cases, a flat set called a Collection MAY be used:
+
+``` ipldsch
+type Collection = [Fact]
+```
+
+# 3 Canonicalization Considerations
+
+IPLD cleanly canonicalizes data, though differently per codec. However, the same data MAY have multiple CIDs due to differences in encoding, hash algorithm, and so on. Strictly speaking, this in no way poses a problem for PomoDB: the same fact entered into the store twice is trivial for operations that only depend on the graph structure of the store.
+
+Certain aggregate functions (e.g. counts, sums, averages) and stateful queries (e.g. graph colorings) depend on a node being present no more than once per graph. Deduplication is thus imperative for many use cases. It is RECOMMENDED that all facts added to a store have a canonical CID. This MAY be of any CID configuration. To reduce the amount of recomputation, using the following parameters are RECOMMENDED:
+
+| Parameter    | Recommended Setting |
+|--------------|---------------------|
+| CID Version  | [CIDv1]             |
+| [Multicodec] | [DAG-CBOR]          |
+| [Multihash]  | [SHA2-256]          |
+
+Note that the [multibase] of a CID is defined by the codec and CID version.
 
 <!-- Links -->
 
+[Attribute]: #213-attribute
 [Brooklyn Zelenka]: https://github.com/expede
+[CIDv1]: https://docs.ipfs.tech/concepts/content-addressing/#cid-versions
+[Capsule Type]: https://notes.brooklynzelenka.com/Capsule+Types
+[Capsule]: #216-capsule
+[Caused By]: #215-caused-by
+[DAG-CBOR]: https://ipld.io/specs/codecs/dag-cbor/spec/
+[DAG-JSON]: https://ipld.io/docs/codecs/known/dag-json/
+[Entity ID]: #212-entity-id
+[Fact]: #211-fact
 [Fission Codes]: https://fission.codes
+[IEEE 754-2019]: https://en.wikipedia.org/wiki/IEEE_754
+[IPFS]: https://ipfs.io
+[IPLD]: https://ipld.io
+[Multicodec]: https://github.com/multiformats/multicodec
+[Multihash]: https://multiformats.io/multihash/ 
```

Comment:
I just realized this link is broken now 😅 It's the same one they link on their homepage, so it's on their end, but I don't know if there's a better place for us to reference? Maybe the github repo, like we do for multicodec?

https://github.com/multiformats/multihash

Reply:
Wow that's not great! That definely loaded yesterday 😬 

Will update; thanks!